src/components/Calendar.tsx

```diff
@@ -23,17 +29,20 @@ class Calendar extends React.PureComponent<CalendarProps, CalendarState> {
     value: this.props.selected
   };
 
-  private onChange = value => {
+  private onChange = (value: CalendarValue) => {
     const { range, onChange } = this.props;
     this.setState(
       {
         value
       },
-      () => (range ? value.length === 2 && onChange(value) : onChange(value))
+      () =>
+        range && Array.isArray(value)
```

Comment:
this is logic change, make sure if range is true and value is not an array it should not call onChange(value)

Reply:
Can we have a test for this ? just this part.